tools/rewriter/GenCallAsm.cpp

```diff
@@ -637,6 +898,45 @@ std::string emit_asm_wrapper(const CAbiSignature &sig,
     |addr |stack by the call to the wrapped function.
     +-----+
 
+    For AArch64:
+    +-----+
+    | top |Top of the stack (stack grows down on AArch64). This address is
+    |     |aligned to 16 bytes.
+    +-----+
+    |ind  |Padding for alignment prior to the indirect args
+    |args |as such memory must be 16B aligned. (TODO: verify this)
+    |align|
+    +-----+
+    |ind  |Space for arguments passed indirectly (i.e. in memory). These arguments
+    |args |will point to this region.
+    +-----+
+    |ret  |Padding for alignment prior to the compartment's return value (if it
+    |align|has class MEMORY) as such memory must be 16B aligned.
+    +-----+
+    |     |Space for the compartment's return value if it has class MEMORY. This
+    |ret  |space is only allocated if the pointer to the caller's return value
+    |space|memory is also placed on the compartment's stack.
+    +-----+
+    |ret  |A pointer saving the caller's address for return value. Only added if
+    |ptr  |using ret space. Saves the previous value of x8.
+    +-----+
+    |     |8 bytes for alignment. If the size of the other items on the stack
+    |     |(including the return address) aren't a multiple 16, these 8 bytes
+    |align|are inserted to ensure the stack is aligned for the call. Otherwise
+    |     |this doesn't get placed on the stack. The System V ABI specifies this
+    |     |must be placed at the end of the argument area so this cannot be
+    |     |placed any lower on the stack. (TODO: verify this)
+    +-----+
+    |stack|Space required for stack arguments. This is initialized from the
+    |args |analogous part of the caller's stack. If all arguments are passed in
+    |     |registers this space isn't allocated.
+    +-----+
+    |frame|Previous link register contents. TODO: update this when we transition stacks?
+    |ptr  |
+    +-----+
+    |ret  |Return address for wrapped function. This is implicitly placed on the
```

Comment:
nit: `bl` on arm64 only updates the link register. it doesn't implicitly put it on the stack right?

Reply:
Also this and the frame ptr blocks are in the wrong place since `emit_prologue` is the first thing called.

Reply:
Ah sorry I meant frame pointer and ret addr pushed on by the prologue of the callee. The frame pointer and return address of the wrapper is on the caller's stack. I'll clarify that this is part of the next frame.